docs/building-a-form/available-form-features-and-usage-guidelines.md

```diff
@@ -448,4 +449,18 @@ If you don't specify a `preSubmitInfo` section, no notice or checkbox appears ab
 
 For the code implementation, see [`PreSubmitSection`](../../src/js/components/PreSubmitSection.jsx) and [`SubmitController`](../../src/js/review/SubmitController.jsx).
 
+### Remote error and event monitoring
+
+If you provide a function for `formConfig.recordEvent`, the library calls that function when notable events occur. If you do not provide a function, the library logs these events onto the browser console.
+
+The `recordEvent` function receives a single object that can contain different information depending on the kind of event being reported. Events reported by the library always have a `event` property that is a string describing the event. The events currently supported by the library are:
+* **validation-failed-on-submit**: This is likely an error in the form definition or a React component, since the form should be completely validated at this point.
+* **form-submit-pending**: An informational message showing that the user has pressed the submit button, and that the form has validated and been sent to the server.
+* **form-submit-successful**: The server returned a status indicating it has accepted the form.
+* **form-submit-error**: The form was submitted, but the server didn't accept the submission. The object contains an `error` and `errorType` with more information about the nature of the error.
+
+### Usage guidelines
+
+Web applications can fail for many reasons, including bad Internet connections, outdated browsers, and misbehaved browser extensions. Even if you test thoroughly, users may experience frustrating errors that they do not report. We highly recommend that you use an error and event reporting service to track the use of your forms. Examples of services that could be used are Google Analytics, Errorception, Sentry, Airbrake, and Raygun.
```

Comment:
What do you think about including a code sample to show what this function might look like? Something like:

"Here's an example of what the function passed to `formConfig.recordEvent` might look like if you were using Google Analytics to track events:

```js
formConfig = {
  ...
  recordEvent: (event) => {
    return window.dataLayer && window.dataLayer.push(event);
  },
  ...
```
"

Reply:
Maybe also an example of how they would change the name of the event they log on their side?